ui/mirage/factories/agent.js

```diff
@@ -8,20 +8,51 @@ const AGENT_STATUSES = ['alive', 'leaving', 'left', 'failed'];
 
 export default Factory.extend({
   id: i => (i / 100 >= 1 ? `${UUIDS[i]}-${i}` : UUIDS[i]),
-  name: () => `nomad@${faker.random.boolean() ? faker.internet.ip() : faker.internet.ipv6()}`,
+  config: makeConfig,
 
-  status: () => faker.helpers.randomize(AGENT_STATUSES),
-  serfPort: () => faker.random.number({ min: 4000, max: 4999 }),
-
-  address() {
-    return this.name.split('@')[1];
-  },
-
-  tags() {
-    const rpcPortCandidate = faker.random.number({ min: 4000, max: 4999 });
+  member() {
     return {
-      port: rpcPortCandidate === this.serfPort ? rpcPortCandidate + 1 : rpcPortCandidate,
-      dc: faker.helpers.randomize(DATACENTERS),
+      name: generateName(),
+      status: faker.helpers.randomize(AGENT_STATUSES),
+      serfPort: faker.random.number({ min: 4000, max: 4999 }),
     };
   },
+
+  afterCreate(agent) {
+    const address = generateAddress.bind(agent);
+    const tags = generateTags.bind(agent);
+    agent.update({
+      member: {
+        ...agent.member,
+        address: address(),
+        tags: tags(),
+      },
+    });
+  },
 });
+
+function generateName() {
```

Comment:
kept these down here and ignored ES6 for the benefit of hoisting to make the file easier to read, do we have opinions on this?

Reply:
I don't have any personal preference, though them seem to be called in line in other factories? https://github.com/hashicorp/nomad/blob/main/ui/mirage/factories/task-group.js#L9

Reply:
These are dynamic attributes which we would normally put in function calls.

tags() {
  return `some${logic}`;
}

However, we're inside of an object, and we no longer get the benefit of automatically running the function when the new instance of this Factory is created.

To me, normally in this case we wouldn't have an object or a list for a property because that would indicate a relationship to another entity, hence why we'd use `trait` or `afterCreate`. I think the pattern here would be consistent because we'd have to use this pattern for `tags` since it's really verbose.